File: Backend/WebAPI/Src/Core/Entities/Comment.cs
namespace Core.Entities;

public class Comment
{
    public Guid Id { get; }
    public Guid BaseUserId { get; set; }
    public Guid LessonId { get; set; }
    public required string Content { get; set; }
    public required DateTime CreatedAt { get; set; }
    public required DateTime UpdatedAt { get; set; }
}

File: Backend/WebAPI/src/main/java/com/vhl/webapi/entities/specific/Comment.java
package com.vhl.webapi.entities.specific;

import com.vhl.webapi.entities.superclass.ICUBaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Table(name = "comments")
public class Comment extends ICUBaseEntity {
    @Column(name = "content", nullable = false, length = 10000)
    private String content;

    @ManyToOne
    @JoinColumn(name = "base_user_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private BaseUser baseUser;

    @ManyToOne
    @JoinColumn(name = "lesson_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Lesson lesson;

    @OneToMany(mappedBy = "fromComment", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Response> responses = new ArrayList<>();
}
